resources/views/windowlight.blade.php

```diff
@@ -85,6 +85,19 @@
                                                 <x-text-input type="text" id="backgroundInput" name="background" value="{{ $background }}" list="colors" title="Enter a valid hexadecimal color code, or leave blank to use a transparent background" placeholder="#FFFFFF" class="w-48" />
                                                 <input type="color" name="backgroundPicker" id="backgroundPicker" value="{{ $background === 'transparent' ? '#ffffff' : $background }}" class="h-auto bg-transparent cursor-pointer ml-2" />
                                             </div>
+                                            <div id="color-presets" class="flex space-x-2 mt-2">
+                                                <button type="button" class="w-6 h-6 rounded-full border border-gray-300" style="background-color: #FFFFFF;" title="White" data-color="#FFFFFF"></button>
+                                                <button type="button" class="w-6 h-6 rounded-full border border-gray-300" style="background-color: #F3F4F6;" title="Light Gray" data-color="#F3F4F6"></button>
+                                                <button type="button" class="w-6 h-6 rounded-full border border-gray-300" style="background-color: #1F2937;" title="Dark Gray" data-color="#1F2937"></button>
+                                                <button type="button" class="w-6 h-6 rounded-full border border-gray-300" style="background-color: #111827;" title="Almost Black" data-color="#111827"></button>
+
+                                                <!-- <button type="button" class="w-6 h-6 rounded-full border border-gray-300 bg-gradient-to-r from-pink-500 via-red-500 to-yellow-500" title="Gradient" data-color="gradient"></button> -->
```

Comment:
Gradient support gets complex with backend support. Something for a future PR. 


Here is a Claude draft

```php
<?php

namespace App\Http\Controllers;

use App\Http\Requests\GenerateRequest;
use App\Models\Analytics\CodeGenerationEvent;
use App\TorchlightSnippetGenerator;
use Illuminate\Http\RedirectResponse;
use Illuminate\View\View;

class WindowlightController extends Controller
{
    public function show(): View
    {
        [$input, $result, $options] = $this->getSessionData();

        [$input, $result] = $this->injectExamplesForEmptyState($input, $result);

        return view('windowlight', array_merge([
            'input' => $input,
            'result' => $result,
            'resultId' => $this->makeResultId($result, $options),
        ], $options));
    }

    public function store(GenerateRequest $request): RedirectResponse
    {
        $validated = $request->validated();

        $request->session()->put('input', $validated['code']);
        $request->session()->put('options.language', $validated['language'] ?? '');
        $request->session()->put('options.lineNumbers', $validated['lineNumbers'] ?? true);
        $request->session()->put('options.background', $validated['background'] ?? 'transparent');
        $request->session()->put('options.useHeader', $validated['useHeader'] ?? true);
        $request->session()->put('options.useShadow', $validated['useShadow'] ?? false);
        $request->session()->put('options.headerButtons', $validated['headerButtons'] ?? true);
        $request->session()->put('options.headerText', $validated['headerText'] ?? '');

        $torchlight = new TorchlightSnippetGenerator(
            $validated['code'],
            $validated['language'] ?? '',
            $validated['lineNumbers'] ?? true
        );

        $result = $torchlight->generate();

        $request->session()->put('result', $result);
        $request->session()->flash('generated');

        CodeGenerationEvent::create([
            'language' => $validated['language'] ?? '',
            'hasMenubar' => (bool) ($validated['useHeader'] ?? true),
            'hasLineNumbers' => (bool) ($validated['lineNumbers'] ?? true),
            'hasMenuButtons' => (bool) ($validated['headerButtons'] ?? true),
            'hasMenubarText' => (bool) ($validated['headerText'] ?? ''),
            'background' => $validated['background'] ?? 'transparent',
            'lines' => substr_count($validated['code'], "\n") + 1,
        ]);

        return redirect()->route('home');
    }

    // ... (rest of the methods remain the same)
}

```

```php
<?php

namespace App\Http\Requests;

use Illuminate\Foundation\Http\FormRequest;
use Illuminate\Support\Str;

class GenerateRequest extends FormRequest
{
    public function authorize(): bool
    {
        return true;
    }

    protected function prepareForValidation(): void
    {
        $this->merge([
            'lineNumbers' => $this->input('lineNumbers') === 'on',
            'useHeader' => $this->input('useHeader') === 'on',
            'headerButtons' => $this->input('headerButtons') === 'on',
            'useShadow' => $this->input('useShadow') === 'on',
            'background' => $this->normalizeBackground($this->input('background') ?: 'transparent'),
        ]);
    }

    public function rules(): array
    {
        return [
            'code' => 'required|string',
            'language' => 'nullable|string',
            'lineNumbers' => 'nullable|boolean',
            'useHeader' => 'nullable|boolean',
            'useShadow' => 'nullable|boolean',
            'headerButtons' => 'nullable|boolean',
            'headerText' => 'nullable|string',
            'background' => [
                'nullable',
                'string',
                function ($attribute, $value, $fail) {
                    if (!$this->isValidBackground($value)) {
                        $fail('The background field format is invalid.');
                    }
                },
            ],
        ];
    }

    protected function normalizeBackground($value): ?string
    {
        if ($value === 'transparent' || $value === 'none') {
            return null;
        }

        if (Str::startsWith($value, '{')) {
            $gradient = json_decode($value, true);
            if (json_last_error() === JSON_ERROR_NONE) {
                return $this->gradientJsonToCss($gradient);
            }
        }

        $colors = [
            'white' => '#ffffff',
            'black' => '#000000',
            'gray' => '#f3f4f6',
        ];

        if (isset($colors[$value])) {
            return $colors[$value];
        }

        if (!str_starts_with($value, '#')) {
            $value = "#$value";
        }

        if (strlen($value) === 4) {
            $value = preg_replace('/^#(.)(.)(.)$/', '#$1$1$2$2$3$3', $value);
        }

        return $value;
    }

    protected function isValidBackground($value): bool
    {
        if ($value === null || $value === 'transparent') {
            return true;
        }

        if (preg_match('/^#[a-f0-9]{6}$/i', $value)) {
            return true;
        }

        if (Str::startsWith($value, '{')) {
            $gradient = json_decode($value, true);
            return $this->isValidGradientJson($gradient);
        }

        return false;
    }

    protected function isValidGradientJson($gradient): bool
    {
        if (!is_array($gradient) || !isset($gradient['type'], $gradient['stops'])) {
            return false;
        }

        if (!in_array($gradient['type'], ['linear', 'radial'])) {
            return false;
        }

        if (!is_array($gradient['stops']) || count($gradient['stops']) < 2) {
            return false;
        }

        foreach ($gradient['stops'] as $stop) {
            if (!isset($stop['color']) || !preg_match('/^#[a-f0-9]{6}$/i', $stop['color'])) {
                return false;
            }
            if (isset($stop['position']) && !preg_match('/^\d+%$/', $stop['position'])) {
                return false;
            }
        }

        return true;
    }

    protected function gradientJsonToCss($gradient): string
    {
        $type = $gradient['type'];
        $direction = $gradient['direction'] ?? ($type === 'linear' ? 'to right' : 'at center center');
        
        $stops = array_map(function($stop) {
            return $stop['color'] . (isset($stop['position']) ? ' ' . $stop['position'] : '');
        }, $gradient['stops']);

        return sprintf('%s-gradient(%s, %s)', $type, $direction, implode(', ', $stops));
    }
}

```